src/amo/components/FeaturedAddons/index.js

```diff
@@ -38,19 +59,39 @@ export class FeaturedAddonsBase extends React.Component {
     }
   }
 
+  loadDataIfNeeded(nextProps = {}) {
+    const { addonType: lastAddonType } = this.props;
+    const {
+      errorHandler, dispatch, loading, results,
+    } = { ...this.props, ...nextProps };
+
+    const nextParams = nextProps && nextProps.params ?
+      nextProps.params : {};
+    const nextAddonType = nextParams.visibleAddonType ?
+      apiAddonType(nextParams.visibleAddonType) : this.requestedAddonType();
+
+    if ((!results || nextAddonType !== lastAddonType) && !loading) {
+      dispatch(getFeatured({
+        addonType: nextAddonType,
+        errorHandlerId: errorHandler.id,
+      }));
+    }
+  }
+
   render() {
-    const { addonType, hasSearchParams, loading, results } = this.props;
+    const { errorHandler, hasSearchParams, loading, results } = this.props;
 
     return (
       <div className="FeaturedAddons">
+        {errorHandler.hasError() ? errorHandler.renderError() : null}
         <h2 className="FeaturedAddons-header">
-          {this.headerForAddonType(addonType)}
+          {this.headerForAddonType()}
```

Comment:
I was gonna ask about this eventually, but I know we do a fair amount of prop passing from `render()` methods to other component class methods. It makes testing easier... or I guess is how we'd setup methods prop injection for testing... but seeing you not use it here makes things cleaner.

I guess in the future we should just read props inside the method itself and not pass them around inside components as there's no need.